### PR TITLE
Feature: Reformat output with color and indention

### PR DESCRIPTION
**Pull request by dantuck**
I was looking to reformat the balances to have a little more formatting and color. This was the first draft of what could be done with coloring which includes when there is unbalanced transactions.

With the coloring I have also included indention of accounts under their account type and removed the duplication of the account types for easier readability. 

rust_ledger-balance-format

### CODE REVIEW

**Comment by ebcrowder** [comment]
This looks great! The indentation makes things a bit more readable and the colors add just enough distinction.

I'd say go ahead and implement this for the other modules, as well. 

Since the format of the output is changing, we'll have to update the tests as they are currently asserting on the CLI's output. Overall, the tests are pretty brittle in this regard because the assertions expect an exact number of spaces in the output, otherwise the test will fail. I never got around to making the tests more flexible and/or meaningful. 

I don't think we need to refactor the tests as part of this work, but wanted to bring it up as a "gotcha" to look out for when you refactor the tests. Let me know if you have any ideas for making the tests more robust.

**Comment by dantuck** [comment]
Glad you like it. I am sure there can be additional updates to the colors and spacing but should be a solid first draft. I will run the tests and get them to pass but while I am in there I'll see if I can find a way to make it less dependent on the spacing.

I'll rename this PR to be more inline to cover all the layout changes rather than specifically covering balances. Might be a few more days before I have the updates.

If getting the tests is going to be easier to just get it to pass with some padding changes I might just do that and then have another PR or discussion later to come up with some ideas to refactor the tests to be offer more coverage.

File: src/accounts.rs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 extern crate serde_yaml;
 
+use colored::*;
```
Comment (dantuck) [inline src/accounts.rs:3]: Decided on the colored crate over term. Seemed to be a bit easier to control color.

**Comment by dantuck** [comment]
Balances and Accounts reporting is pretty straight forward on how it should be displayed but when it comes to register we are now only allowing for using an account and offset account per transaction so no splitting. I have an idea of how it should look which is included in the code update here. This will look a lot like ledger but will give us control over time as additional feature come out.

What do you think?

image

File: tests/cli.rs [hunk 2]
```diff
@@ -182,7 +181,7 @@ fn register_test() -> Result<(), Box<dyn std::error::Error>> {
         .arg("register")
         .arg("-f=credit_card");
     cmd.assert().success().stdout(predicate::str::contains(
-        "2019-01-01 | 1          | expense-test-acct    | credit_card",
+        "2019-01-01 test memo               expense-test-acct              1           1",
```
Comment (dantuck) [inline tests/cli.rs:184]: Definitely not the best way to do this but it works right now to pass the test. It doesn't test the second line but I have not put any thought into how to do the right testing.

**Comment by dantuck** [comment]
Let me know your thoughts and if we need to connect and work through some of this together I am open for it.

**Comment by ebcrowder** [comment]
This looks great! I like the addition of balance checks on the `register` output, as well. 

In case folks want to disable color output, it looks like the crate checks for whether the `NO_COLOR` env variable is set. If that indeed works, perhaps we could update the `README` to add a line of instructions around that. I could see disabling color output being useful for a lot of scenarios, so we should definitely highlight that in the documentation. I think we'll be good to merge once this is in place.

I definitely punted on making the tests useful when initially building this tool. I'll do some thinking on this so that we don't need to update them each time we change something - but maybe it is a problem we can't avoid? I'll look at other CLIs to see how they handle tests. 

Overall, I could see `register` being the one area that could use a lot of extending to meet parity with the original ledger CLI. My simple use cases don't warrant the full capabilities of ledger, but it would be a fun challenge. I'd be curious to learn what other ideas you have for extending stuff, as well. I could also set up a Riot/Matrix room for collaboration. Let me know!

**Comment by dantuck** [comment]
Glad you like it. I have pushed some changes to the README to include more explanation of the environment variables and also restructured the file so that it is broken up at the top for installation and then usage. If you don't like it I can change that section back.

I totally understand the testing hold back. When I went in to fix the tests I found it best fitting to just change the expected output and then let the test pass. Maybe there is a better way to do checking that the api is working how we expect it but not sure how that would be done yet.

So far I think the app covers the basic use cases so I am now looking into some of the feature that I use most from Ledger. I really like your idea of setting up a collaborative chat room. I have Riot.im installed on most of my devices and that will work out well. Maybe we can take those conversations there and we can connect over email until that is connected. Exciting!

**Comment by ebcrowder** [comment]
Thanks for these changes! Everything looks good to me.

Also, I set up a public Matrix room at: https://matrix.to/#/!dYISGJYNNiZcUrxhcm:matrix.org?via=matrix.org

I'll update the `README` to advertise it, as well.

**Comment by ebcrowder** [comment]
Also, I will publish a new version of the crate tonight so that the PRs you have submitted so far will be reflected.

**Comment by dantuck** [comment]
Thanks. Glad you like it.

**Pull request merged**